lightning/src/routing/scoring.rs

```diff
@@ -623,6 +623,35 @@ impl<G: Deref<Target = NetworkGraph>, L: Deref, T: Time> ProbabilisticScorerUsin
 		assert!(self.channel_liquidities.insert(short_channel_id, liquidity).is_none());
 		self
 	}
+
+	/// Dump the contents of this scorer into the configured logger.
+	///
+	/// Note that this writes roughly one line per channel for which we have a liquidity estimate,
+	/// which may be a substantial amount of log output.
+	pub fn debug_log_liquidity_stats(&self) {
+		let graph = self.network_graph.read_only();
+		for (scid, liq) in self.channel_liquidities.iter() {
+			if let Some(chan_debug) = graph.channels().get(scid) {
+				let log_direction = |dir: &Option<ChannelUpdateInfo>, sender, recipient| {
+					if let Some(dir_debug) = dir {
+						let amount = chan_debug.capacity_sats.map(|v| v*1000).or(dir_debug.htlc_maximum_msat);
+						if let Some(amt) = amount {
+							let dir_liq = liq.as_directed(sender, recipient, amt, self.params.liquidity_offset_half_life);
+							log_debug!(self.logger, "Liquidity from {:?} to {:?} via {} is in the range ({}, {})",
+								sender, recipient, scid, dir_liq.min_liquidity_msat(), dir_liq.max_liquidity_msat());
+						} else {
+							log_debug!(self.logger, "No amount known for SCID {} from {:?} to {:?}", scid, sender, recipient);
+						}
```

Comment:
You should be able to use `chan_debug.as_directed_to(...).effective_capacity().as_msat()`. This will also handle the `EffectiveCapacity::Unknown` case.